client/src/components/atoms/ImagePicker.tsx

```diff
@@ -1,24 +1,167 @@
+}
 const ImagePicker: FunctionComponent<Props> = (props: Props) => {
-    const { selected, images, onImageChange, isErroneous } = props;
+    const [crop, setCrop] = useState({ x: 0, y: 0 });
+    const [zoom, setZoom] = useState(1);
+    const [showWarning, setShowWarning] = useState(false);
+    const [warningText, setWarningText] = useState("");
+    const { selected, images, onImageChange, isErroneous, onUploadImg, uploadedImg } = props;
+
+    const getImageDimensions = async (file: string): Promise<Dimensions> => {
+        return new Promise((resolved) => {
+            const i = new Image();
+            i.onload = () => {
+                resolved({ width: i.width, height: i.height });
+            };
+            i.src = file;
+        });
+    };
+    const handleOnDrop = (acceptedFiles: File[]) => {
+        const file = acceptedFiles[0];
+        let validFile = true;
+        if (file.size > 6000000) {
+            triggerWarning("Images must be less than 5MB");
+            validFile = false;
+        }
+        if (file.type != "image/png" && file.type != "image/jpeg") {
+            triggerWarning("Images must be in either JPEG or PNG format");
+            validFile = false;
+        }
+        const fileReader = new FileReader();
+        fileReader.addEventListener(
+            "load",
+            async () => {
+                const res = fileReader.result;
+                console.log(res);
+                let imgStr;
+                if (res === null) imgStr = "";
+                else if (typeof res === "string") {
+                    imgStr = res;
+                } else {
+                    imgStr = res.toString();
+                }
+                const dimensions = await getImageDimensions(imgStr);
+                console.log(dimensions);
+                if (dimensions.width < 600 || dimensions.height < 430) {
+                    triggerWarning("Images must be at least 600 x 430 pixels");
+                    validFile = false;
+                }
+                if (validFile) {
+                    setShowWarning(false);
+                    onUploadImg(imgStr);
+                }
+            },
+            false
+        );
+        fileReader.readAsDataURL(file);
+    };
+
+    const clearImage = () => {
+        onUploadImg("");
+        onImageChange("");
+    };
+
+    const changeZoom = (increment: boolean) => {
+        if (increment) {
+            if (zoom >= 10) return;
+            setZoom(zoom + 0.1);
+        } else {
+            if (zoom <= 1) return;
+            setZoom(zoom - 0.1);
+        }
+    };
+    const triggerWarning = (text: string) => {
+        setShowWarning(true);
+        setWarningText(text);
+        setTimeout(() => {
+            setShowWarning(false);
+        }, 4000);
+    };
+    const imgInView = selected !== "" || uploadedImg !== "";
     return (
         <div className="imagepicker">
+            <WarningBox text={warningText} showWarning={showWarning} />
```

Comment:
Could we conditionally render `WarningBox` instead of passing a prop?
```
{showWarning && <WarningBox text={warningText}  />}
```

Reply:
Updated. I also changed the rendering condition to be if the warningText is empty instead of having a separate state as suggested by Philips below.